### PR TITLE
feat: server info module (#22)

### PR DESCRIPTION
**Pull request by avirtopeanu-ionos**
## What does this fix or implement?

(#22)

Added module to get all servers for a given datacenter id.

Use optional param `upgrade_needed` (boolean) to only list those servers that need / don't need an upgrade

This PR also has some out-of-scope fixes:
    - made s3key_info only require token or username and password for auth
    - fixed s3key_info name (for info modules should be plural): S3 Key to S3 Keys
    - added test for server module in compute-engine/all-tests
    - removed duplicate `lan` param for server module

## Checklist

- [x] PR name added as appropriate (e.g. feat/fix/doc/test/refactor/etc)
- [x] Documentation updated
- [x] Jira task updated
- [x] Wrote tests

### CODE REVIEW

**Review by mihaela-mh-ionos — APPROVED** [review]
Amazing work 🥇 
My only suggestion is to link the corresponding issue on this PR (#22) - for transparency

**Review by AnaMariaMoga — APPROVED** [review]

File: plugins/modules/server_info.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    - name: Get all servers for given datacenter
+      server_info:
+        datacenter: "{{ datacenter }}"
+        register: server_list_response
+
+    - name: Get only the servers that need to be upgraded
+      server_info:
+        datacenter: "{{ datacenter }}"
+        register: servers_list_upgrade_response
+        upgradeNeeded: true
```
Comment (Raduionutz) [inline plugins/modules/server_info.py:109]: I think this should be snake case

Comment (avirtopeanu-ionos) [reply plugins/modules/server_info.py:109]: good catch, now i also saw that register should be 1 tab to the left, thank you

Comment (Raduionutz) [reply plugins/modules/server_info.py]: you're welcome

**Pull request merged**